### DISCUSSION TITLE
Dataloader support for mutations

### DISCUSSION THREAD

**Discussion by palhoye** [graphql-java/java-dataloader#164] [General]
As mentioned [here](https://www.graphql-java.com/documentation/batching/#data-loader-only-works-with-asyncexecutionstrategy), dataloader only works with `AsyncExecutionStrategy`. 

Since mutations use `AsyncSerialExecutionStrategy` ([see code](https://github.com/graphql-java/graphql-java/blob/d6dbf6156e8a90a1420a7cee957de0b22100d47f/src/main/java/graphql/GraphQL.java#L293)), it means that dataloader does not work with mutations.

Could you please confirm this is the case, and if so, if there are plans to add dataloader support for mutation responses? 

Thanks, I truly appreciate the good work you are doing with graphql-java and dataloaders!

**Comment by rstoyanchev** [comment]
@palhoye I didn't notice that there is a link to create an issue from a discussion. At any rate, I created https://github.com/graphql-java/graphql-java/issues/3711, and it's reasonable to expect the behavior to be clarified at least in the documentation, if indeed there is a reason why it cannot work.
